eth_utils/abi.py

```diff
@@ -1,29 +1,67 @@
+from collections import (
+    abc,
+)
+import copy
+import itertools
+import re
 from typing import (
     Any,
     Dict,
+    Iterable,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    Tuple,
+    Union,
+    cast,
+)
+
+from eth_typing import (
+    ABI,
+    ABIComponent,
+    ABIElement,
+    ABIError,
+    ABIEvent,
+    ABIFallback,
+    ABIFunction,
+    ABIReceive,
+)
+from typing_extensions import (
+    deprecated,
+)
+
+from eth_utils.types import (
+    is_list_like,
 )
 
 from .crypto import (
     keccak,
 )
 
 
+@deprecated(
+    "`collapse_if_tuple(abi)` is deprecated in favor of"
+    "`get_normalized_abi_arg_type(abi)`"
+)
 def collapse_if_tuple(abi: Dict[str, Any]) -> str:
     """
     Converts a tuple from a dict to a parenthesized list of its types.
```

Comment:
The deprecated decorator doesn't give any indication in the docs that  the method is deprecated. Maybe add a note there too?